### PR TITLE
PDF -  implement height calculation for cells with rowSpan (with wordwrap) 

### PR DESCRIPTION
**Pull request by novsstation**

### CODE REVIEW

File: js/exporter/jspdf/v3/height_updater.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+            .filter(cell => isDefined(cell.rowSpan));
+
+        cellsWithRowSpan.forEach(cell => {
+            const cellTextHeight = calculateTextHeight(doc, cell.pdfCell.text, cell.pdfCell.font, {
+                wordWrapEnabled: cell.pdfCell.wordWrapEnabled,
+                columnWidth: cell.pdfCell._rect.w
+            });
+
+            const rowsCount = cell.rowSpan + 1;
+            const summaryRowsHeight = rows
```
Comment (IgnatovDan) [inline js/exporter/jspdf/v3/height_updater.js]: summaryRowsHeight -> currentRowSpanRowsHeight

File: js/exporter/jspdf/v3/pdf_utils_v3.js [hunk 2]
```diff
@@ -31,9 +31,13 @@ function calculateRowHeight(doc, cells, columnWidths) {
 
     let rowHeight = 0;
     for(let cellIndex = 0; cellIndex < cells.length; cellIndex++) {
-        const cellText = cells[cellIndex].text;
-        const font = cells[cellIndex].font;
-        const wordWrapEnabled = cells[cellIndex].wordWrapEnabled;
+        if(isDefined(cells[cellIndex].rowSpan)) {
```
Comment (IgnatovDan) [inline js/exporter/jspdf/v3/pdf_utils_v3.js:34]: It's not clear whether cells[cellIndex] is 'mergedCellsMaster' or 'mergedCellsSlave' ?
Does it make sense in this condition ?

**Review by IgnatovDan — CHANGES_REQUESTED** [review]

File: js/exporter/jspdf/v3/pdf_utils_v3.js (hunk 2 above)
Comment (novsstation) [reply js/exporter/jspdf/v3/pdf_utils_v3.js:34]: > Does it make sense in this condition ?

Nope, I skip all cells with rowsSpan, including mergedCellsMaster and mergedCellsSlave.
This code computes height only for simple cells without any rowSpan.

Height for the cells with rowSpan will be computer later.
I will resize all rows to fit such rows cell with rowSpans

File: js/exporter/jspdf/v3/height_updater.js (hunk 1 above)
Comment (novsstation) [reply js/exporter/jspdf/v3/height_updater.js]: yep, great idea - fixed

**Review by KuznetsovVN — APPROVED** [review]

**Review by IgnatovDan — APPROVED** [review]

**Pull request merged**